adapta/storage/distributed_object_store/v3/datastax_astra/astra_client.py

```diff
@@ -324,33 +321,33 @@ def to_frame(
                 else num_threads
             )
             with ThreadPoolExecutor(max_workers=max_threads) as tpe:
-                result = concat(
-                    tpe.map(
-                        lambda args: to_frame(*args),
-                        [
-                            (cassandra_model, key_column_filter, select_columns)
-                            for key_column_filter in compiled_filter_values
-                        ],
-                        chunksize=max(int(len(compiled_filter_values) / num_threads), 1),
-                    )
+                data = tpe.map(
+                    lambda args: apply(*args),
+                    [
+                        (cassandra_model, key_column_filter, select_columns)
+                        for key_column_filter in compiled_filter_values
+                    ],
+                    chunksize=max(int(len(compiled_filter_values) / num_threads), 1),
                 )
         else:
-            result = concat(
-                [
-                    MetaFrame(
-                        [dict(v.items()) for v in list(apply(cassandra_model, key_column_filter, select_columns))],
-                        convert_to_polars=(lambda x: polars.DataFrame(x, schema=select_columns))
-                        if not deduplicate
-                        else (lambda x: polars.DataFrame(x, schema=select_columns).unique()),
-                        convert_to_pandas=(lambda x: pandas.DataFrame(x, columns=select_columns))
-                        if not deduplicate
-                        else (lambda x: pandas.DataFrame(x, columns=select_columns).drop_duplicates()),
-                    )
-                    for key_column_filter in compiled_filter_values
-                ]
+            data = (
+                apply(cassandra_model, key_column_filter, select_columns)
+                for key_column_filter in compiled_filter_values
             )
 
-        return result
+        data = itertools.chain.from_iterable(data)
+        if limit:
```

Comment:
hmm but we did apply the limit in the query already? why do we apply it twice here?